invui-core/src/main/java/xyz/xenondevs/invui/item/Item.java

```diff
@@ -21,6 +21,18 @@ public interface Item {
      */
     ItemProvider getItemProvider();
```

Comment:
```suggestion
    default ItemProvider getItemProvider() {
        throw new UnsupportedOperationException("getItemProvider() or getItemProvider(Player) must be overridden!");
    }
```

What do you think about making this a default method as well to prevent people from having to implement `getItemProvider()` as well if they override `getItemProvider(Player)`

Reply:
Not entirely sure if this is abi compatible though

Reply:
I did consider this - and in a perfect world I'd prefer that to be the case - however I was concerned about this breaking existing usage/implementation (e.g. cases where users are calling `Item#getItemProvider` directly - for whatever reason) as this could lead to UOEs occurring in edge cases.
I don't know how extensive an impact that may be though - it could certainly be worth trying and if it becomes an issue, just swap to the delegation route instead?

Reply:
For that to happen, the user would need to actively choose to use `getItemProvider(Player)` in one of their items, at which point they should be aware that the other method will throw, no?

Reply:
Actually yeah, that's a really good point.
If they're implementing Item/AbstractItem directly they already must be overriding getItemProvider, that would still work the same and the abstractions within InvUI would still be fine too
That's an oversight on my part!
I've pushed up a change to the PR to throw a UOE now as you suggested instead :))